src/mqueryfront/src/QueryMonaco.js

```diff
@@ -0,0 +1,163 @@
+import React, { Component } from "react";
+import Editor, { monaco } from "@monaco-editor/react";
+import YARA from "./yara-lang";
+
+class QueryMonaco extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            rawYara: props.rawYara,
+            editor: null,
+            readOnly: false,
+            decorations: [],
+        };
+        this.handleEditorDidMount = this.handleEditorDidMount.bind(this);
+    }
+
+    getValue() {
+        if (this.state.editor) {
+            return this.state.editor.getValue();
+        }
+    }
+
+    setValue(newValue) {
+        if (this.state.editor) {
+            this.state.editor.setValue(newValue);
+        }
+    }
+
+    setReadOnly(readOnly) {
+        if (this.state.editor) {
+            this.setState({ readOnly: readOnly });
+        }
+    }
+
+    setError(error) {
```

Comment:
Maybe we should unpack error to several arguments instead of treating it like unlabeled array. This will make setError more generic and ready for further changes.
```suggestion
    setError(error, startLine, startColumn, endColumn, errorMessage) {
```
We can call it using spread syntax in the regex part. Maybe we should include the actual errorMessage as well in case we want to show some tooltips?
```javascript
this.refs.editor.setError(
    ...err.match(/Error at (\d+).(\d+)-?(\d+)?: (.*)/)
);
```